react-carousel/my-app/src/Carousel.tsx

```diff
@@ -0,0 +1,91 @@
+import { useEffect, useState } from 'react';
+import {
+  FaAngleLeft,
+  FaAngleRight,
+  FaCircle,
+  FaRegCircle,
+} from 'react-icons/fa';
+import { type Image } from './App';
+import './Carousel.css';
+
+type Props = {
+  images: Image[];
+};
+
+export function Carousel({ images }: Props) {
+  const [imageIndex, setImageIndex] = useState(0);
+
+  useEffect(() => {
+    const timerId = setTimeout(() => {
+      setImageIndex((imageIndex + 1) % images.length);
+    }, 1000);
+    return () => clearTimeout(timerId);
+  }, [setImageIndex, imageIndex, setTimeout, clearTimeout]);
```

Comment:
If you watch the typescript-closures Q&A you'll see that you don't need to put several of these into the dependency array. The only things that need to be in the dependency array are things that can change.
`setTimeout` and `clearTimeout` are JavaScript global functions, so their values will never change.
`setImageIndex` is a React state setter, and React guarantees that their values will not change.

So the only thing you need in the dependency array is `imageIndex`!